distributed/shuffle/tests/test_shuffle.py

```diff
@@ -654,10 +657,68 @@ def test_processing_chain():
+    df = pd.DataFrame(
+        {
+            f"col{next(counter)}": pd.array([True, False] * 50, dtype="bool"),
+            f"col{next(counter)}": pd.array([True, False] * 50, dtype="boolean"),
+            f"col{next(counter)}": pd.array(range(100), dtype="int8"),
+            f"col{next(counter)}": pd.array(range(100), dtype="int16"),
+            f"col{next(counter)}": pd.array(range(100), dtype="int32"),
+            f"col{next(counter)}": pd.array(range(100), dtype="int64"),
+            f"col{next(counter)}": pd.array(range(100), dtype="uint8"),
+            f"col{next(counter)}": pd.array(range(100), dtype="uint16"),
+            f"col{next(counter)}": pd.array(range(100), dtype="uint32"),
+            f"col{next(counter)}": pd.array(range(100), dtype="uint64"),
+            f"col{next(counter)}": pd.array(range(100), dtype="float16"),
+            f"col{next(counter)}": pd.array(range(100), dtype="float32"),
+            f"col{next(counter)}": pd.array(range(100), dtype="float64"),
+            # f"col{next(counter)}": pd.array(range(100), dtype="csingle"),
+            # f"col{next(counter)}": pd.array(range(100), dtype="cdouble"),
+            # f"col{next(counter)}": pd.array(range(100), dtype="clongdouble"),
+            f"col{next(counter)}": pd.array(
+                [np.datetime64("2022-01-01")] * 100, dtype="datetime64"
+            ),
+            f"col{next(counter)}": pd.array(
+                [np.datetime64("2022-01-01")] * 100,
+                dtype=pd.DatetimeTZDtype(tz="UTC"),
+            ),
+            f"col{next(counter)}": pd.array(
+                [np.timedelta64(1, "D")] * 100, dtype="timedelta64"
+            ),
+            f"col{next(counter)}": pd.array(
+                [pd.Period("2022-01-01", freq="D")] * 100, dtype="period[D]"
+            ),
+            f"col{next(counter)}": pd.array(
+                [pd.Interval(left=0, right=5)] * 100, dtype="Interval"
+            ),
+            f"col{next(counter)}": pd.array(range(100), dtype="Int8"),
+            f"col{next(counter)}": pd.array(range(100), dtype="Int16"),
+            f"col{next(counter)}": pd.array(range(100), dtype="Int32"),
+            f"col{next(counter)}": pd.array(range(100), dtype="Int64"),
+            f"col{next(counter)}": pd.array(range(100), dtype="UInt8"),
+            f"col{next(counter)}": pd.array(range(100), dtype="UInt16"),
+            f"col{next(counter)}": pd.array(range(100), dtype="UInt32"),
+            f"col{next(counter)}": pd.array(range(100), dtype="UInt64"),
+            f"col{next(counter)}": pd.array(["x", "y"] * 50, dtype="category"),
+            # f"col{next(counter)}": pd.array(
+            # [np.nan, np.nan, 1.0, np.nan, np.nan] * 20,
+            # dtype="Sparse[float64]",
+            # ),
+            f"col{next(counter)}": pd.array(["lorem ipsum"] * 100, dtype="string"),
+            # f"col{next(counter)}": pd.array(
+            #     [Stub(i) for i in range(100)], dtype="object"
+            # ),
```

Comment:
raises `pyarrow.lib.ArrowInvalid: Could not convert <distributed.shuffle.tests.test_shuffle.test_processing_chain.<locals>.Stub object at 0x11f3fe820> with type Stub: did not recognize Python value type when inferring an Arrow data type`